Column.indices may contain the index from another schema

If there is an index with the same table name and column name in another schema, the index of the other schema is acquired by `extractSchemas`.

### Reproducible code: 

```ts

    test("it should extract indices", async ({ knex: [db] }) => {
      await db.raw(`
        create table test.some_table (id integer);
        create index some_table_id_idx on test.some_table (id);
        create schema test2;
        create table test2.some_table (id integer);
        create index some_table_id_idx2 on test2.some_table (id);
      `);

      const result = await extractTable(db, makePgType("some_table"));

      const expected: Index[] = [
        {
          isPrimary: false,
          name: "some_table_id_idx",
        },
      ];

      expect(result.columns[0].indices).toStrictEqual(expected);
    });
```


### Expected behavior:

The test passes.

### Actual behavior:

The test fails because the `indices` contains the index from another schema:

```
AssertionError: expected [ { …(2) }, { …(2) } ] to strictly equal [ { isPrimary: false, …(1) } ]

- Expected
+ Received

  Array [
    Object {
      "isPrimary": false,
      "name": "some_table_id_idx",
    },
+   Object {
+     "isPrimary": false,
+     "name": "some_table_id_idx2",
+   },
  ]
```

### Version:

Reproduced in the latest main branch.

--- a/src/kinds/extractTable.test.ts
+++ b/src/kinds/extractTable.test.ts
@@ -4,6 +4,7 @@ import { describe, expect } from "vitest";
 import { test, testWith } from "../tests/useSchema";
 import extractTable, {
   ColumnReference,
+  Index,
   TableCheck,
   TableColumn,
   TableDetails,
@@ -392,6 +393,27 @@ describe("extractTable", () => {
         onUpdate: "NO ACTION",
         name: "fk_1",
       });
+    });
+
+    test("should not extract indices from another schema", async ({
+      knex: [db],
+    }) => {
+      await db.raw("create table test.some_table (id integer)");
+      await db.raw("create index some_table_id_idx on test.some_table (id)");
+      await db.raw("create schema test2");
+      await db.raw("create table test2.some_table (id integer)");
+      await db.raw("create index some_table_id_idx2 on test2.some_table (id)");
+
+      const result = await extractTable(db, makePgType("some_table"));
+
+      const expected: Index[] = [
+        {
+          isPrimary: false,
+          name: "some_table_id_idx",
+        },
+      ];
+
+      expect(result.columns[0].indices).toStrictEqual(expected);
     });
   });
 });

--- a/src/kinds/query-parts/indexMapQueryPart.ts
+++ b/src/kinds/query-parts/indexMapQueryPart.ts
@@ -10,14 +10,17 @@ const indexMapQueryPart = `
     pg_class t,
     pg_class i,
     pg_index ix,
-    pg_attribute a
+    pg_attribute a,
+    pg_namespace n
   WHERE
     t.oid = ix.indrelid
     AND i.oid = ix.indexrelid
     AND a.attrelid = t.oid
+    AND n.oid = t.relnamespace
     AND a.attnum = ANY (ix.indkey)
     AND t.relkind = 'r'
     AND t.relname = :table_name
+    AND n.nspname = :schema_name
   GROUP BY
     a.attname
 `;

